#ifndef GUI_STORAGES_WIDGET_H
#define GUI_STORAGES_WIDGET_H

////////////////////////////////////////////////////////////////////////////////
//
// Includes
//
#ifndef DB_MANAGER_H
#	include "db_manager.h"
#endif
#ifndef DB_STORAGES_DATA_H
#	include "db_storagesdata.h"
#endif

#include "ui_storages.h"

// Qt includes
#include <QWidget>
#include <memory>
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
//
// class CStoragesWidget
//
class CStoragesWidget : public QWidget
{
public:// Constructors
	CStoragesWidget(QWidget* pwParent = nullptr);
	~CStoragesWidget() = default;

public:// Interface Methodes

protected:// Helper Methodes

private:
	Ui::storages m_uiStorages;

	//QDialog* m_pCreateSplDlg;

	std::shared_ptr<db::CStoragesData> m_pStoragesData;
	std::shared_ptr<db::CDBManager> m_pDBManager;

};
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////

#endif // GUI_STORAGES_WIDGET_H